### ISSUE TITLE
Fix dropzone issues

### ISSUE THREAD

**Issue by gsidebo** [mitodl/bootcamp-ecommerce#901]
There are a few issues with the dropzone library we're using (). 

#### Accessibility

From Alice:
> the react-dropzone-uploader library we're using is not keyboard accessible. see this issue: fortana-co/react-dropzone-uploader#10

We probably need to supply a custom component to the library to fix this ([doc](https://react-dropzone-uploader.js.org/docs/customization#default-components))

#### React warning
A react warning comes up in the console when you open the resume drawer

> Each child in a list should have a unique "key" prop. See https://fb.me/react-warning-keys for more information.
    in label
    ...
    in AppContainer

Similar to the accessibility issue, we probably need to supply a custom component to the library to avoid this ([doc](https://react-dropzone-uploader.js.org/docs/customization#default-components))

#### Message clarity
- If an upload fails, it's not clear to the user that anything happened
- There is no text indication that says only PDFs are allowed
- If a user tries to upload a non-PDF, the file isn't uploaded, but no message is shown

### Acceptance Criteria
- Dropzone inputs are accessible
- Dropzone no longer causes React warning in locally running app
- Error message is clearer when an upload fails
- Message added to indicate that only PDF files are accepted
- Some error/indicator shown when a user tries to upload a non-PDF file

**Comment by aliceriot** [comment]
for the input on the dropzone I think I found a CSS only way to do it. Basically, input elements should normally be tab-able, but if they have `display: none` set on them they get taken out of the tab order. So basically I removed `display: none` and simulated it by using `position: fixed` and `opacity: 0` - this makes the input tab-able

**Comment by gsidebo** [comment]
@aliceriot Regarding error messages when something fails to upload: This library does not give you a way to access the server response from a file upload request. Someone opened a PR to add this feature – _gulp_ – almost a year ago, but it still hasn't been reviewed or merged by the library author(s): https://github.com/fortana-co/react-dropzone-uploader/pull/34. Judging from the comments it seems like it works. We could potentially use a fork of that repo with that feature implemented (or create our own fork), but obviously we'll need to consider whether or not we'll want to replace this library since it seems to be abandoned by its maintainer. We get a bunch of features out-of-the-box with this library that aren't available in `react-dropzone`, so we need to consider that.

**Comment by mbertrand** [comment]
There is also a '[Unexpected token =>](https://sentry.io/organizations/mit-office-of-digital-learning/issues/1796812276/events/?environment=production&project=159445&referrer=alert_email)' error that seems to occur primarily on mobile phones.

**Comment by aliceriot** [comment]
yeah, so that is coming from a really pretty old mobile phone, looks like it's running android 5.1.1 (from ~2014) and Samsung Internet v 3.5. I think right now in most of our projects we are not compiling our JS down to ES5 but we're targeting environments that support ES6. If we wanted to support phones like this we could change our babel transpilation target to be ES5 JavaScript and it should run. Alternatively, if we get questions from users trying to use the site on such an old phone we could recommend that they install a more modern browser, since even if it's not possible to upgrade the phone OS it should still be possible to install a more recent version of Chrome from the Play store.

**Comment by aliceriot** [comment]
hmm so actually that's confusing, because afaict our setup should be compiling down to ES5 currently. I think there could be an issue where there's untranspiled code being included from `node_modules`, but not sure.
